src/ocm/components/hosts/use-hosts-table.tsx

```diff
@@ -161,23 +160,23 @@ export const useHostsTable = (cluster: Cluster) => {
   const onUpdateDay2ApiVip: UpdateDay2ApiVipFormProps['onUpdateDay2ApiVip'] = React.useCallback(
     async (apiVip: string, onError: (message: string) => void) => {
       try {
-        const { data } = await ClustersAPI.update(cluster.id, {
+        const { data } = await ClustersService.update(cluster, {
           apiVipDnsName: apiVip,
         });
         dispatch(updateCluster(data));
       } catch (e) {
         handleApiError(e, () => onError(getApiErrorMessage(e)));
       }
     },
-    [cluster.id, dispatch],
+    [cluster, dispatch],
```

Comment:
I don't like that now we have a dependency on the entire `cluster` object, which means that on each polling, the effects will have to be recalculated. This could affect performance.

Also, I see a problem that now the UI is in charge of not overwriting tags that we didn't create, and in theory we could run into race conditions were someone is setting a tag while someone else is on the UI, and some could be lost.

And it can happen the other way around, the users could do some changes on the UI, but later they decide to tag them via the cli with an automated script for example, and the UI tag could be lost.

So I'd prefer an API which lets add "add/remove" a tag, rather than patching the tags to "set" them.

Reply:
@jgyselov I understand now that we are aware of this limitation from the API and that we are OK with it.
Also, speaking to @jkilzi he mentioned that through tools like Kibana we would have the entire history of requests for a specific server so that would identify 100% of the clusters that at some point sent the OCM_UI tag.